internal_ws/ykcompile/src/arch/x86_64/mod.rs

```diff
@@ -615,7 +599,9 @@ impl TraceCompiler {
         let sym_addr = if let Some(addr) = self.addr_map.get(sym) {
             *addr as i64
         } else {
-            TraceCompiler::find_symbol(sym)? as i64
+            // This path is required for system calls. Currently hwtracer will never trace through
+            // the kernel, so system call addresses will never appear in addr_map.
+            find_symbol(sym)? as i64
```

Comment:
I know this is the `x86_64` file, but I was curious why `i64` here and not another integer type (`usize` or `u64`).

Reply:
I think dynasm requires it to be signed.

Reply:
Yes.

```
  624 |             ; mov Rq(*TEMP_REG), QWORD sym_addr
      |                                        ^^^^^^^^ expected `i64`, found `u64`
```

This is probably leakage from some x86_64 encoding constraint.